perceval/utils/statevector.py

```diff
@@ -78,6 +78,19 @@ def allstate_iterator(input_state: Union[BasicState, StateVector], mask=None) ->
         for output_state in output_array:
             yield BasicState(output_state)
 
+def bs_iterator(m: int, n_max: int):
+    """
+    Iterator on all possible output state on m modes with at most n_max photons
+
+    :param m: number of modes
+    :param n_max: maximum number of photons
+    :return: list of BasicState
+    """
+    for n in range(n_max+1):
+        output_array = FSArray(m,n)
+        for output_state in output_array:
+            yield BasicState(output_state)
```

Comment:
Can't I also get undirectly the whole list with this syntaxe ? I felt like it allowed more flexibility because I could directly iterate on the function...

Reply:
You can. It's as you wish.

Reply:
I think i'm confortable to use it like that. Moreover it is consistent with the other iterator.